pkg/kubectl/cmd/apiresources.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+var (
+	apiresources_example = templates.Examples(`
+		# Print the supported API Resources
+		kubectl api-resources`)
+)
+
+type APIResourceOptions struct {
+	APIGroup    string
+	APIResource metav1.APIResource
+}
+
+func NewCmdApiResources(f cmdutil.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "api-resources",
+		Short:   "Print the supported API resources on the server",
+		Long:    "Print the supported API resources on the server",
+		Example: apiresources_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(RunApiResources(f, out))
```

Comment:
I'd like to see the ability to scope this to a specific group or group/version. In the meantime, please verify no additional arguments are passed (`len(args) == 0`), which will let us take filtering args in the future